worlds/dark_souls_3/__init__.py

```diff
@@ -302,30 +303,31 @@ def create_items(self) -> None:
         # Gather all default items on randomized locations
         self.local_itempool = []
         num_required_extra_items = 0
-        for location in self.multiworld.get_unfilled_locations(self.player):
+        for location in cast(List[DarkSouls3Location], self.multiworld.get_unfilled_locations(self.player)):
             if not self._is_location_available(location.name):
                 raise Exception("DS3 generation bug: Added an unavailable location.")
 
-            item = item_dictionary[location.data.default_item_name]
+            default_item_name = cast(str, location.data.default_item_name)
+            item = item_dictionary[default_item_name]
             if item.skip:
                 num_required_extra_items += 1
             elif not item.unique:
-                self.local_itempool.append(self.create_item(location.data.default_item_name))
+                self.local_itempool.append(self.create_item(default_item_name))
             else:
                 # For unique items, make sure there aren't duplicates in the item set even if there
                 # are multiple in-game locations that provide them.
-                if location.data.default_item_name in item_set:
+                if default_item_name in item_set:
                     num_required_extra_items += 1
                 else:
-                    item_set.add(location.data.default_item_name)
-                    self.local_itempool.append(self.create_item(location.data.default_item_name))
+                    item_set.add(default_item_name)
+                    self.local_itempool.append(self.create_item(default_item_name))
 
         injectables = self._create_injectable_items(num_required_extra_items)
         num_required_extra_items -= len(injectables)
         self.local_itempool.extend(injectables)
 
         # Extra filler items for locations containing skip items
-        self.local_itempool.extend(self.create_filler() for _ in range(num_required_extra_items))
+        self.local_itempool.extend(cast(DarkSouls3Item, self.create_filler()) for _ in range(num_required_extra_items))
```

Comment:
Probably better to declare `self.create_filler()` as returning a `DarkSouls3Item`

Reply:
You'd have to add that, since it isn't defined, but we could?